### PR TITLE
Kb/hq workflow changes

### PR DESCRIPTION
**Pull request by kbo001**
## Summary
Implemented execution of the HQSmokeTests scripts on production as well during pushes

## Description
Added a sequential job of executing the HQSmokeTests scripts on Production as well when we push any changes.

## QA Checklist

- [x] Label(s) and Assignee added
- [X] PR sent for review
- [ ] Documentation (if applicable) added/updated

### CODE REVIEW

**Comment by snopoke** [comment]
I think (untested) that this should work from looking at the docs:
* https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs
* https://docs.github.com/en/actions/learn-github-actions/expressions#fromjson
* https://docs.github.com/en/actions/using-jobs/defining-outputs-for-jobs

I've kept the actual build job very basic so that we could see if this works. Once it's working we can add the rest of the build steps.

```yaml
jobs:
  set_matrix:
    runs-on: ubuntu-latest
    outputs:
      matrix: ${{ steps.set-matrix.outputs.matrix || steps.set-matrix-default.outputs.matrix }}
    steps:
      - id: set-matrix
        if: ${{ github.event.client_payload.environment != null }}
        run: echo "::set-output name=matrix::{\"environment\": [\"${{ github.event.client_payload.environment }}\"]}"
      - id: set-matrix-default
        if: ${{ github.event.client_payload.environment == null }}
        run: echo "::set-output name=matrix::{\"environment\": [\"production\", \"staging\"]}"
  build:
    needs: set_matrix
    strategy:
      matrix: ${{fromJSON(needs.set_matrix.outputs.matrix)}}
    name: Smoke on '${{ matrix.environment }}'
    runs-on: ubuntu-latest
    steps:
      - run: |
        echo "client_payload: ${{ toJson(github.event.client_payload) }}"
        echo "matrix environment: ${{ matrix.environment }}"
```

**Review by snopoke — APPROVED** [review]
Looks great.

File: .github/workflows/hq-smoke-tests.yml [hunk 1]
```diff
@@ -12,120 +12,128 @@ on:
...
-      uses: actions/upload-artifact@v2
-      with:
-        name: test-result-reports
-        path: /home/runner/work/dimagi-qa/dimagi-qa/report.html
-        retention-days: 2
+      - name: Archive test results
+        if: always()
+        uses: actions/upload-artifact@v2
+        with:
+          name: test-result-reports
```
Comment (snopoke) [inline .github/workflows/hq-smoke-tests.yml:137]: I think this needs to include the env name to make it unique per run

Comment (kbo001) [reply .github/workflows/hq-smoke-tests.yml]: done!

File: .github/workflows/hq-smoke-tests.yml [hunk 2]
```diff
@@ -12,120 +12,128 @@ on:
...
-      with:
-        name: test-result-reports
-        path: /home/runner/work/dimagi-qa/dimagi-qa/report.html
-        retention-days: 2
+      - name: Archive test results
+        if: always()
+        uses: actions/upload-artifact@v2
+        with:
+          name: test-result-reports-${{ matrix.environment }}
+          path: /home/runner/work/dimagi-qa/dimagi-qa/report_${{ matrix.environment }}.html
```
Comment (snopoke) [inline .github/workflows/hq-smoke-tests.yml:138]: Rather use the provided workspace path
```suggestion
          path: ${{ github.workspace }}/report_${{ matrix.environment }}.html
```

**Review by snopoke — APPROVED** [review]

File: .github/workflows/hq-smoke-tests.yml [hunk 3]
```diff
@@ -12,120 +12,128 @@ on:
...
+      matrix: ${{ steps.set-matrix.outputs.matrix || steps.set-matrix-default.outputs.matrix }}
+    steps:
+      - id: set-matrix
+        if: ${{ github.event.client_payload.environment != null }}
+        run: |
+          echo "::set-output name=matrix::{\"environment\": [\"${{ github.event.client_payload.environment }}\"]}"
+      - id: set-matrix-default
+        if: ${{ github.event.client_payload.environment == null }}
+        run: |
+          echo "::set-output name=matrix::{\"environment\": [\"production\", \"staging\"]}"
```
Comment (snopoke) [inline .github/workflows/hq-smoke-tests.yml:28]: @kbo001 is it really necessary to run on both envs for every push / PR?

**Pull request merged**